src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala

```diff
@@ -110,16 +109,16 @@ class TasksResource @Inject() (
     @QueryParam("scale")@DefaultValue("false") scale: Boolean,
     @QueryParam("force")@DefaultValue("false") force: Boolean,
     body: Array[Byte],
-    @Context req: HttpServletRequest, @Context resp: HttpServletResponse): Response = {
+    @Context req: HttpServletRequest): Response = {
 
     val taskIds = (Json.parse(body) \ "ids").as[Set[String]]
-    val taskToAppIds = taskIds.map { id =>
+    val tasksToAppId = taskIds.map { id =>
       try { id -> Task.Id.appId(id) }
       catch { case e: MatchError => throw new BadRequestException(s"Invalid task id '$id'.") }
     }.toMap
 
-    doIfAuthorized(req, resp, KillTask, taskToAppIds.values.toSeq: _*) { implicit identity =>
-
+    val affectedApps = tasksToAppId.values.flatMap(appId => result(groupManager.app(appId))).toSeq
```

Comment:
A lot of blocking calls to groupManager. Use `groupManager.root` and have all apps.
Other option: move the auth calls to the TaskKiller